Code/Framework/AzFramework/AzFramework/Quality/QualityCVarGroup.cpp

```diff
@@ -0,0 +1,322 @@
+
+        // handle integer values (fast path)
+        QualityLevel qualityLevel = FromNumber(value);
+
+        if (qualityLevel == QualityLevel::Invalid)
+        {
+            // handle enum pre-defined values 
+            qualityLevel = FromEnum(value);
+        }
+
+        if (qualityLevel == QualityLevel::Invalid)
+        {
+            // handle user-defined level name values
+            qualityLevel = FromName(value);
+        }
+
+        return qualityLevel;
+    }
+
+    QualityLevel QualityCVarGroup::FromNumber(const AZ::CVarFixedString& value) const
+    {
+        char* endPtr = nullptr;
+        int32_t intValue = aznumeric_cast<int32_t>(strtoll(value.c_str(), &endPtr, 0));
+
+        // don't allow values higher than the user-defined levels
+        if (endPtr == value.c_str() || intValue >= m_numQualityLevels)
+        {
+            return QualityLevel::Invalid;
+        }
+
+        return QualityLevel(intValue);
+    }
+
+    QualityLevel QualityCVarGroup::FromEnum(const AZ::CVarFixedString& value) const
+    {
+        using EnumMemberPair = typename AZ::AzEnumTraits<QualityLevel>::MembersArrayType::value_type;
+        auto findEnumOptionValue = [firstArg = value](EnumMemberPair enumPair)
+        {
+            // case-insensitive comparison
+            constexpr bool caseSensitive{ false };
+            return AZ::StringFunc::Equal(firstArg, enumPair.m_string, caseSensitive);
+        };
+        if (auto foundIt = AZStd::ranges::find_if(AZ::AzEnumTraits<QualityLevel>::Members,
+            findEnumOptionValue); foundIt != AZ::AzEnumTraits<QualityLevel>::Members.end())
+        {
+            return foundIt->m_value;
+        }
+
+        return QualityLevel::Invalid;
+    }
+
+    QualityLevel QualityCVarGroup::FromName(const AZ::CVarFixedString& value) const
+    {
+        using FixedValueString = AZ::SettingsRegistryInterface::FixedValueString;
+        using VisitResponse = AZ::SettingsRegistryInterface::VisitResponse;
+
+        auto registry = AZ::SettingsRegistry::Get();
+        if (registry == nullptr)
+        {
+            return QualityLevel::Invalid;
+        }
+
+        QualityLevel qualityLevel{ QualityLevel::Invalid };
+
+        // visit each user-defined level name and perform a case-insensitive comparison
+        int32_t arrayIndex{ 0 };
+        auto callback = [&](const auto& visitArrayArg)
+        {
+            FixedValueString qualityLevelName;
+            if (registry->Get(qualityLevelName, visitArrayArg.m_jsonKeyPath))
+            {
+                // case-insensitive comparison
+                constexpr bool caseSensitive{ false };
+                if (AZ::StringFunc::Equal(qualityLevelName, value, caseSensitive))
+                {
+                    qualityLevel = QualityLevel(arrayIndex);
+                    return VisitResponse::Done;
+                }
+            }
+            arrayIndex++;
+            return VisitResponse::Continue;
+        };

```

Comment:
FYI there is a way to add mark lambda members as mutable and to declare new members using the capture list.
The [mutable](https://en.cppreference.com/w/cpp/language/lambda#Explanation) keyword can be used to have capture value types be non-const.
You can combine that with using an identifier in a [capture](https://en.cppreference.com/w/cpp/language/lambda#Lambda_capture) to initialize the arrayIndex member to 0.
```suggestion
        // visit each user-defined level name and perform a case-insensitive comparison
        auto callback = [&, arrayIndex = 0](const auto& visitArrayArg) mutable
        {
            FixedValueString qualityLevelName;
            if (registry->Get(qualityLevelName, visitArrayArg.m_jsonKeyPath))
            {
                // case-insensitive comparison
                constexpr bool caseSensitive{ false };
                if (AZ::StringFunc::Equal(qualityLevelName, value, caseSensitive))
                {
                    qualityLevel = QualityLevel(arrayIndex);
                    return VisitResponse::Done;
                }
            }
            arrayIndex++;
            return VisitResponse::Continue;
        };

```

Reply:
I like this idea, though in this case I think it'll be simpler for devs who are less familiar with these newer features to read the current code.  Just my preference really though, not a strong one.